Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,682 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using MediaBrowser.Common;
+using MediaBrowser.Common.Extensions;
+using MediaBrowser.Common.Json;
+using MediaBrowser.Common.Json.Converters;
+using MediaBrowser.Common.Plugins;
+using MediaBrowser.Model.Configuration;
+using MediaBrowser.Model.Plugins;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Emby.Server.Implementations.Plugins
+{
+    /// <summary>
+    /// Defines the <see cref="PluginManager" />.
+    /// </summary>
+    public class PluginManager : IPluginManager
+    {
+        private readonly string _pluginsPath;
+        private readonly Version _appVersion;
+        private readonly JsonSerializerOptions _jsonOptions;
+        private readonly ILogger<PluginManager> _logger;
+        private readonly IApplicationHost _appHost;
+        private readonly ServerConfiguration _config;
+        private readonly IList<LocalPlugin> _plugins;
+        private readonly Version _minimumVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PluginManager"/> class.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/>.</param>
+        /// <param name="appHost">The <see cref="IApplicationHost"/>.</param>
+        /// <param name="config">The <see cref="ServerConfiguration"/>.</param>
+        /// <param name="pluginsPath">The plugin path.</param>
+        /// <param name="appVersion">The application version.</param>
+        public PluginManager(
+            ILogger<PluginManager> logger,
+            IApplicationHost appHost,
+            ServerConfiguration config,
+            string pluginsPath,
+            Version appVersion)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _pluginsPath = pluginsPath;
+            _appVersion = appVersion ?? throw new ArgumentNullException(nameof(appVersion));
+            _jsonOptions = JsonDefaults.GetOptions();
```

Comment:
With the changes from https://github.com/jellyfin/jellyfin/pull/4855 you must now use the copy constructor if you need to make any changes to the options

Reply:
Thanks